### ISSUE TITLE
Using the = sign with user formula causes a crash.

### ISSUE THREAD

**Issue by dillon101001** [xaos-project/XaoS#88]
Entering custom formulae that are equalities cause an instant crash. 
Examples would be 3=3, z+i=c, and c+2=c.
It is understandable that xaos wouldn't be able to handle these, but it shouldn't crash from them.

**Comment by jblang** [comment]
Reproduced on master. Valgrind stack trace:

```
==24815== Invalid read of size 8
==24815==    at 0x18083C: sffe_parse (sffe.c:856)
==24815==    by 0x126A9F: uih_sffein (menu.c:1407)
==24815==    by 0x112CD6: ui_menuactivate (ui.c:415)
==24815==    by 0x11F208: ui_closedialog (dialog.c:1358)
==24815==    by 0x11F208: ui_dialogkeys (dialog.c:1452)
==24815==    by 0x112E79: ui_key (ui.c:840)
==24815==    by 0x17D33C: x11_processevents (ui_x11.c:314)
==24815==    by 0x110577: ui_mainloop (ui.c:1668)
==24815==    by 0x110577: main (ui.c:1482)
==24815==  Address 0x72a05c0 is 16 bytes before a block of size 0 alloc'd
==24815==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==24815==    by 0x17FD4B: sffe_parse (sffe.c:655)
==24815==    by 0x126A9F: uih_sffein (menu.c:1407)
==24815==    by 0x112CD6: ui_menuactivate (ui.c:415)
==24815==    by 0x11F208: ui_closedialog (dialog.c:1358)
==24815==    by 0x11F208: ui_dialogkeys (dialog.c:1452)
==24815==    by 0x112E79: ui_key (ui.c:840)
==24815==    by 0x17D33C: x11_processevents (ui_x11.c:314)
==24815==    by 0x110577: ui_mainloop (ui.c:1668)
==24815==    by 0x110577: main (ui.c:1482)
==24815== 
```

**Comment by jblang** [comment]
This should be fixed with d7563bcf1fc24496bbdd9a5ee4201a68eab30102, which fixes detection of unrecognized operators, among other things.
